lib/type-checker.js

```diff
@@ -398,10 +393,20 @@ class TypeChecker {
   }
 
   _stringTypes(string, highPrecisionSupport) {
+    let types = [ 'String', 'Object', 'Array' ];
```

Comment:
looks like this module is using `var` everywhere else, maybe for backwards compatibility? should we just follow that pattern?